source/wp-content/themes/wporg-developer-2023/patterns/cli-commands.php

```diff
@@ -7,24 +7,55 @@
 
 ?>
 
-<!-- wp:group {"align":"wide","style":{"spacing":{"padding":{"left":"var:preset|spacing|edge-space","right":"var:preset|spacing|edge-space","top":"var:preset|spacing|60","bottom":"var:preset|spacing|60"}}},"layout":{"type":"constrained"}} -->
-<div class="wp-block-group alignwide" style="padding-top:var(--wp--preset--spacing--60);padding-right:var(--wp--preset--spacing--edge-space);padding-bottom:var(--wp--preset--spacing--60);padding-left:var(--wp--preset--spacing--edge-space)"><!-- wp:columns {"align":"wide"} -->
-<div class="wp-block-columns alignwide"><!-- wp:column {"width":"33%"} -->
-<div class="wp-block-column" style="flex-basis:33%"><!-- wp:heading {"level":1,"fontSize":"heading-2"} -->
-<h1 class="wp-block-heading has-heading-2-font-size">WP-CLI Commands</h1>
-<!-- /wp:heading --></div>
-<!-- /wp:column -->
+<!-- wp:group {"tagName":"main","style":{"spacing":{"padding":{"left":"var:preset|spacing|edge-space","right":"var:preset|spacing|edge-space","bottom":"var:preset|spacing|60"}}},"className":"alignfull","layout":{"type":"constrained","wideSize":"1280px","contentSize":"680px"}} -->
+<main class="wp-block-group alignfull" style="padding-right:var(--wp--preset--spacing--edge-space);padding-bottom:var(--wp--preset--spacing--60);padding-left:var(--wp--preset--spacing--edge-space)"><!-- wp:group {"align":"wide","style":{"spacing":{"margin":{"top":"var:preset|spacing|20","bottom":"var:preset|spacing|20"}}},"layout":{"type":"constrained","justifyContent":"left"}} -->
+<div class="wp-block-group alignwide" style="margin-top:var(--wp--preset--spacing--20);margin-bottom:var(--wp--preset--spacing--20)"><!-- wp:search {"label":"<?php esc_attr_e( 'Search', 'wporg' ); ?>","showLabel":false,"placeholder":"<?php esc_attr_e( 'Search resources', 'wporg' ); ?>","width":240,"widthUnit":"px","buttonText":"Search","buttonPosition":"button-inside","buttonUseIcon":true} /--></div>
+<!-- /wp:group -->
+
+<!-- wp:group {"align":"wide","style":{"spacing":{"margin":{"top":"var:preset|spacing|20"}}},"layout":{"type":"constrained","justifyContent":"left"}} -->
+<div class="wp-block-group alignwide" style="margin-top:var(--wp--preset--spacing--20)"><!-- wp:group {"tagName":"article","style":{"spacing":{"blockGap":"var:preset|spacing|40","margin":{"top":"0px"}}}} -->
+<article class="wp-block-group" style="margin-top:0px"><!-- wp:wporg/sidebar-container -->
+<!-- wp:group -->
+<div class="wp-block-group"><!-- wp:wporg/table-of-contents /--></div>
```

Comment:
This block expects to be able to get the content of the page using the post id in the block context. Because this is an archive template and posts are queried, the post id in the context is that of the first command post, not the page. Need to find a solution for getting the template content without creating an infinite loop.

Reply:
sounds like the main query needs resetting - do you know how the posts are queried for the commands?

Reply:
There isn't a post here to get the content from, the headings here are hardcoded in the pattern.

Could you use a styled list block? it's a little duplicate work to get the styles right, but that's probably better than trying to use the dynamic block. I guess it wouldn't collapse, but there are only two headings. hmm… Does the core table of contents block help? No, it's not dynamic at all.

There's the `wporg_table_of_contents_post_content` filter, if you could use that somehow to "load" in the pattern content, then we could probably remove the `! isset( $block->context['postId'] )` check in the ToC block?

Reply:
> There isn't a post here to get the content from, the headings here are hardcoded in the pattern.

Very good point 😂 

Thanks, I'll have a play with these ideas.

Reply:
> There's the `wporg_table_of_contents_post_content` filter, if you could use that somehow to "load" in the pattern content

This works nicely, see https://github.com/WordPress/wporg-developer/pull/288/commits/74d955d69e9bea632555e017c94521e87de1903b where I split out the content to be parsed (static content minus the h1) and feed it through the filter.